extension/options.js

```diff
@@ -1,29 +1,50 @@
 window.addEventListener( 'load', function () {
-  var localStorageKey = 'format_preferences';
-
-  var manifest = chrome.runtime.getManifest();;
+  var manifest = chrome.runtime.getManifest();
   var version = document.getElementById("version");
   version.innerText = manifest.version;
 
   chrome.runtime.getBackgroundPage(function (backgroundWindow) {
-    var formats = backgroundWindow.instance().formats;
+    var target = backgroundWindow.instance();
+    var formats = target.formats;
+
+    var defaultFormatButton = document.getElementById('set-default-format');
+    var currentDefault = target.getDefaultFormat();
+    document.getElementById('current-default-format').textContent = String(currentDefault);
+    defaultFormatButton.textContent = 'Set default to ' + currentDefault;
+    defaultFormatButton.addEventListener('click', function () {
+      var v = defaultFormatButton.getAttribute('data-selected-value');
+      backgroundWindow.instance().setDefaultFormat(v);
+      document.getElementById('current-default-format').textContent = v;
+    });
     
     try{
-      var ctable	= new CocoaTable(formats, [
+      var ctable = new CocoaTable(formats, [
         'label', 'format', 'filter'
       ] );
+      
+      ctable.onSelectedRowChanged(function (rowEvent) {
+        try {
+          var currentSelectedRow = rowEvent.detail.current;
+          var selectedName = currentSelectedRow.querySelector('span').textContent;
+          defaultFormatButton.textContent = 'Set default to ' + selectedName;
+          defaultFormatButton.setAttribute('data-selected-value', selectedName);
+        } catch (e) {
+          console.log(e);
```

Comment:
This would happen while the user is in the middle of editing/creating a new format. Safe to ignore, so we do.